### PR TITLE
fix: add the logic of removing the finalizer in yurt-iot-dock

### LINKED ISSUE #1694: [BUG] Lack of the delete logic of finalizer in yurt-iot-dock
**What happened**:
Can not delete the device on the openyurt platform. The reason is the finalizer in the device crd.
**What you expected to happen**:
Delete the redundantKubeDevices in the sync time loop.
**How to reproduce it (as minimally and precisely as possible)**:
Create a device and then delete the device. But it can't be deleted in the edgex platform.
**Anything else we need to know?**:
I find the solution and I want to fix this bug.

**Environment**:
- OpenYurt version: master
- Kubernetes version (use `kubectl version`):
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**others**

/kind bug

### PR DESCRIPTION
**Pull request by Pluviophile225**
#### What type of PR is this?
/kind bug
/sig iot

#### What this PR does / why we need it:
add the logic of removing the finalizer
#### Which issue(s) this PR fixes:

Fixes #1694 

#### Special notes for your reviewer:

/assgin @LavenderQAQ @wangxye

### CODE REVIEW

File: pkg/yurtiotdock/controllers/deviceservice_syncer.go [hunk 1]
```diff
@@ -197,6 +199,14 @@ func (ds *DeviceServiceSyncer) deleteDeviceServices(redundantKubeDeviceServices
 				"DeviceService", kds.Name)
 			return err
 		}
+		patchData, _ := json.Marshal(map[string]interface{}{
+			"metadata": map[string]interface{}{
+				"finalizers": []string{},
+			},
+		})
+		if err := ds.Client.Patch(context.TODO(), kds, client.RawPatch(types.MergePatchType, patchData)); err != nil {
+			return err
```
Comment (LavenderQAQ) [inline pkg/yurtiotdock/controllers/deviceservice_syncer.go:208]: Add the log before returning the error:
```go
klog.V(5).ErrorS(err, "fail to remove finalizer of DeviceService on Kubernetes", "DeviceService", kds.Name)
```

File: pkg/yurtiotdock/controllers/deviceprofile_syncer.go [hunk 2]
```diff
@@ -213,6 +215,14 @@ func (dps *DeviceProfileSyncer) deleteDeviceProfiles(redundantKubeDeviceProfiles
 				"DeviceProfile", kdp.Name)
 			return err
 		}
+		patchData, _ := json.Marshal(map[string]interface{}{
+			"metadata": map[string]interface{}{
+				"finalizers": []string{},
+			},
+		})
+		if err := dps.Client.Patch(context.TODO(), kdp, client.RawPatch(types.MergePatchType, patchData)); err != nil {
+			return err
```
Comment (LavenderQAQ) [inline pkg/yurtiotdock/controllers/deviceprofile_syncer.go:224]: Add the log before returning the error:
```go
klog.V(5).ErrorS(err, "fail to remove finalizer of DeviceProfile on Kubernetes", "DeviceProfile", dps.Name)
```

File: pkg/yurtiotdock/controllers/device_syncer.go [hunk 3]
```diff
@@ -198,6 +200,14 @@ func (ds *DeviceSyncer) deleteDevices(redundantKubeDevices map[string]*iotv1alph
 				"DeviceName", kd.Name)
 			return err
 		}
+		patchData, _ := json.Marshal(map[string]interface{}{
+			"metadata": map[string]interface{}{
+				"finalizers": []string{},
+			},
+		})
+		if err := ds.Client.Patch(context.TODO(), kd, client.RawPatch(types.MergePatchType, patchData)); err != nil {
+			return err
```
Comment (LavenderQAQ) [inline pkg/yurtiotdock/controllers/device_syncer.go:209]: Add the log before returning the error:
```go
klog.V(5).ErrorS(err, "fail to remove finalizer of Device on Kubernetes", "Device", ds.Name)
```

**Review by LavenderQAQ — CHANGES_REQUESTED** [review]

**Review by LavenderQAQ — APPROVED** [review]
/lgtm

**Comment by LavenderQAQ** [comment]
PTAL @wangxye

**Review by rambohe-ch — APPROVED** [review]

**Pull request merged**